filewatcher/filewatcher.go

```diff
@@ -161,24 +202,18 @@ func (r *Result) watcherLoop(
 				continue
 			}
 
-			if filepath.Base(ev.Name) != file {
-				continue
-			}
-
 			switch ev.Op {
 			default:
 				// Ignore uninterested events.
-			case fsnotify.Create, fsnotify.Write:
-				mtime, err := getMtime(path)
-				if err != nil {
-					logger.Log(context.Background(), fmt.Sprintf(
-						"filewatcher: failed to get mtime for %q: %v",
-						path,
-						err,
-					))
-					continue
+			case fsnotify.Create, fsnotify.Write, fsnotify.Rename, fsnotify.Remove:
+				// Use fsEventDelay to avoid calling forceReload repetively when a burst
+				// of fs events happens (for example, when multiple files within the
+				// directory changed).
+				if timer == nil {
+					timer = time.AfterFunc(fsEventsDelay, forceReload)
```

Comment:
there's still a race condition here it looks like... if it takes longer than 200ms for the files to get written, and then you have an error watching the file(s), I think it can cause the watcher to effectively "forget" about the files forever.  It doesn't currently look to me like the polling interval will recover, though I may be missing code that would make it do so.

Reply:
we only remove and readd watched files after parser returned success, we also log and ignore all failures of adding files to watch, so failure to watch one file would not affect other files (but yes that particular file could be forgotten forever).

>if it takes longer than 200ms for the files to get written

The intended usage of filewatcher is really for those rename swaps, not really open the files one-by-one and write the updates (we can probably add some more documents to clarify on that). and if someone really is using it to watch directories that's updated one-by-one, the delay is still configurable so they can configure it to be a larger number to mitigate those risks.

Reply:
So the problem that I forsee is that this is used to monitor a CSI driver mount point, and those currently (and for a long time since they haven't even fixed it yet) rm the files before writing the new ones, it seems _very_ easy for a lot of or large files to exceed 200ms.

Reply:
yea, the k8s implementation is not great :/

do you want to make the default longer? I do not want to make it too long mainly to not make the `secrets` test to sleep for too long (we currently don't expose the api to customize it so it has to use the default one). but maybe 1s is not too bad?